nixos/modules/services/web-apps/libretranslate.nix

```
@@ -0,0 +1,203 @@
+        ];
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    environment.systemPackages = [ ltmanageKeysCli ];
+
+    systemd.tmpfiles.rules = lib.mkIf (cfg.dataDir != "/var/lib/libretranslate") [
+      "d '${cfg.dataDir}' 0750 ${cfg.user} ${cfg.group} - -"
+      "z '${cfg.dataDir}' 0750 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.libretranslate = {
+      description = "LibreTranslate service";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      environment = {
+        HOME = cfg.dataDir;
+      };
+      serviceConfig = lib.mkMerge [{
+        Type = "simple";
+        ExecStart = ''
+          ${cfg.package}/bin/libretranslate \
+            --host ${cfg.host} \
+            --port ${toString cfg.port} \
+            ${lib.optionalString cfg.enableApiKeys "--api-keys"} \
+            ${lib.optionalString cfg.disableWebUI "--disable-web-ui"} \
+            ${lib.optionalString cfg.updateModels "--update-models"} \
+            ${lib.optionalString (cfg.threads != null) "--threads ${toString cfg.threads}"} \
+            ${lib.concatStringsSep " " cfg.extraArgs}
+        '';
+        WorkingDirectory = cfg.dataDir;
+        # User and group
+        User = cfg.user;
+        Group = cfg.group;
+        # Proc filesystem
+        ProcSubset = "all";
+        ProtectProc = "invisible";
+        # Access write directories
+        UMask = "0027";
+        # Capabilities
+        CapabilityBoundingSet = "";
+        # Security
+        NoNewPrivileges = true;
+        # Sandboxing
+        ProtectSystem = "strict";
+        ProtectHome = true;
+        PrivateTmp = true;
+        PrivateDevices = true;
+        PrivateUsers = true;
+        ProtectHostname = true;
+        ProtectClock = true;
+        ProtectKernelTunables = true;
+        ProtectKernelModules = true;
+        ProtectKernelLogs = true;
+        ProtectControlGroups = true;
+        RestrictAddressFamilies = [ "AF_INET" "AF_INET6" ];
+        RestrictNamespaces = true;
+        LockPersonality = true;
+        MemoryDenyWriteExecute = false;
+        RestrictRealtime = true;
+        RestrictSUIDSGID = true;
+        RemoveIPC = true;
+        PrivateMounts = true;
+        # System Call Filtering
+        SystemCallArchitectures = "native";
+        SystemCallFilter = [ "~@cpu-emulation @debug @ipc @keyring @mount @obsolete @privileged @setuid" ];
+      }
+      (lib.mkIf (cfg.dataDir == "/var/lib/libretranslate") {
+        # State directory and mode
+        StateDirectory = "libretranslate";
+        StateDirectoryMode = "0750";
+      })
+      (lib.mkIf (cfg.dataDir != "/var/lib/libretranslate") {
+        ReadWritePaths = cfg.dataDir;
+      })];
+    };
+
+    services.nginx = lib.mkIf cfg.configureNginx {
+      enable = true;
+      virtualHosts."${cfg.domain}" = {
+        root = "/var/empty";
+
+        locations."/" = {
+          proxyPass = "http://127.0.0.1:${toString(cfg.port)}";
```

```suggestion
          proxyPass = "http://127.0.0.1:${toString cfg.port}";
```
Can we use a socket instead?

---

The program does not support the use of sockets.